vortex-array/src/compute/search_sorted.rs

```diff
@@ -30,31 +40,106 @@ pub fn search_sorted<T: Into<Scalar>>(
         })
 }
 
-pub trait SearchSorted<T> {
-    fn search_sorted(&self, value: &T, side: SearchSortedSide) -> usize;
+pub trait IndexOrd<Idx: ?Sized, V> {
```

Comment:
I think you want something more like:

```rust
trait IndexCmp {
   fn cmp(&self, a: usize, b: usize) -> Ordering;
}

```

Reply:
I don't think this works with the signature of search_sorted of array which takes a value